SingularityService/src/main/java/com/hubspot/singularity/data/SingularityValidator.java

```diff
@@ -549,4 +561,30 @@ public void checkRequestGroup(SingularityRequestGroup requestGroup) {
 
     checkBadRequest(requestGroup.getRequestIds() != null, "requestIds cannot be null");
   }
+
+  public void checkValidShellCommand(final SingularityShellCommand shellCommand) {
+    Optional<ShellCommandDescriptor> commandDescriptor = Iterables.tryFind(uiConfiguration.getShellCommands(), new Predicate<ShellCommandDescriptor>() {
+      @Override
+      public boolean apply(ShellCommandDescriptor input) {
+        return input.getName().equals(shellCommand.getName());
+      }
+    });
+
+    if (!commandDescriptor.isPresent()) {
+      throw WebExceptions.forbidden("Shell command %s not in %s", shellCommand.getName(), uiConfiguration.getShellCommands());
```

Comment:
I can update that. FYI, this was copy-pasted from the original shell command code in TaskResource, I just moved it to the validator instead